[civiremote_entity] Rebuild form in case of error

dontub opened the pull request:
Previously on error a redirect to the destination (URL parameter) was executed, if given. Otherwise, a redirect to the form page was executed. Now in both cases the form is rebuild (so the entered values are kept).

jensschuppe reviewed (commented):
Will the destination parameter be lost entirely or just "postponed" to the next submit?

Wondering whether we should instead [rebuild](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Form%21FormState.php/property/FormState%3A%3Arebuild/11.x) the form (as failed validation does) - this would avoid a redirect to the destination as well and keep the form instance alive (see [Drupal Form API Workflow](https://www.drupal.org/docs/drupal-apis/form-api/form-api-workflow)).

Commit:
[civiremote_entity] Rebuild form in case of error

Previously on error a redirect to the destination (URL parameter) was
executed, if given. Otherwise, a redirect to the form page was executed.
Now in both cases the form is rebuild (so the entered values are kept).

dontub commented:
You're right, rebuilding is the better approach here so I changed that.
